app/controllers/admin/merchants_controller.rb

```diff
@@ -6,4 +6,23 @@ def index
   def show
     @merchant = Merchant.find(params[:id])
   end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    if @merchant.update(merchant_params)
+      redirect_to "/admin/merchants/#{@merchant.id}"
+      flash[:notice] = "Merchant #{@merchant.id} has been successfully updated"
+    else
+      redirect_to "/admin/merchants/#{@merchant.id}/edit"
```

Comment:
Since you've already implemented a flash message, you could later make a flash message saying all fields have to be filled out. Just an idea